uaclient/config.py

```diff
@@ -751,7 +751,104 @@ def _attached_status(self) -> Dict[str, Any]:
                 response["contract"]["tech_support_level"] = supportLevel
         return response
 
-    def status(self, show_beta=False) -> Dict[str, Any]:
+    def _get_entitlement_information(
+        self, entitlements: List[Dict[str, Any]], entitlement_name: str
+    ) -> Dict[str, Any]:
+        """Extract information from the entitlements array."""
+        for entitlement in entitlements:
+            if entitlement.get("type") == entitlement_name:
+                return {
+                    "entitled": "yes" if entitlement.get("entitled") else "no",
+                    "auto_enabled": "yes"
+                    if entitlement.get("obligations", {}).get(
+                        "enableByDefault"
+                    )
+                    else "no",
+                    "affordances": entitlement.get("affordances", {}),
+                }
+        return {"entitled": "no", "auto_enabled": "no", "affordances": {}}
+
+    def _simulate_status(self, token: str) -> Dict[str, Any]:
+        """Return a status dictionary based on a token."""
+        from uaclient.contract import (
+            get_available_resources,
+            get_contract_information,
+        )
+        from uaclient.entitlements import ENTITLEMENT_CLASS_BY_NAME
+
+        contract_information = get_contract_information(self, token)
+
+        contract_info = contract_information.get("contractInfo", {})
+        account_info = contract_information.get("accountInfo", {})
+
+        response = {
+            "version": version.get_version(features=self.features),
+            "contract": {
+                "id": contract_info.get("id", ""),
+                "name": contract_info.get("name", ""),
+                "created_at": contract_info.get("createdAt", ""),
+                "products": contract_info.get("products", []),
+            },
+            "account": {
+                "name": account_info.get("name", ""),
+                "id": account_info.get("id"),
+                "created_at": account_info.get("createdAt", ""),
+                "external_account_ids": account_info.get(
+                    "externalAccountIDs", []
+                ),
+            },
```

Comment:
Do we need to surface this information ? It seems we just want to present service information in that command

Reply:
I think that basic contract and account information may also be useful for people consuming this endpoint. At installation time, when providing a token, the user can verify that this data is right in the step where there are the services for picking, and also have easy access to this data to report if something goes wrong.
From the complexity side of things, the endpoint in the Contract Server is providing this data directly, so I think it's not bad to include it.

Reply:
Aye, I agree and think that the output of this command should be as close the output of a normal status command as possible.